base/src/main/java/io/spine/base/Environment.java

```diff
@@ -21,37 +21,76 @@
 package io.spine.base;
 
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
 import io.spine.annotation.SPI;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static io.spine.base.BaseEnvironmentType.PRODUCTION;
+import static io.spine.base.BaseEnvironmentType.TESTS;
+
 /**
  * Provides information about the environment (current platform used, etc.).
+ *
+ * <p>By default, knows only about {@link BaseEnvironmentType}. Library users may extend the list
```

Comment:
`BaseEnvironmentType` is a package-private enum. `Environment` is a public class. Something isn't right.

I wouldn't metion that at all.